dipy/io/tests/test_io_peaks.py

```diff
@@ -116,6 +112,33 @@ def test_io_peaks():
         os.path.isfile(fname_gfa)
 
 
-if __name__ == '__main__':
+def test_io_save_peaks_error():
+    with InTemporaryDirectory():
+        fname = 'test.pam5'
+
+        pam = PeaksAndMetrics()
+
+        npt.assert_raises(IOError, save_peaks, 'test.pam', pam)
+        npt.assert_raises(ValueError, save_peaks, fname, pam)
 
+        sphere = get_sphere('repulsion724')
+
+        pam.affine = np.eye(4)
+        pam.peak_dirs = np.random.rand(10, 10, 10, 5, 3)
+        pam.peak_values = np.zeros((10, 10, 10, 5))
+        pam.peak_indices = np.zeros((10, 10, 10, 5))
+        pam.shm_coeff = np.zeros((10, 10, 10, 45))
+        pam.sphere = sphere
+        pam.B = np.zeros((45, sphere.vertices.shape[0]))
+        pam.total_weight = 0.5
+        pam.ang_thr = 60
+        pam.gfa = np.zeros((10, 10, 10))
+        pam.qa = np.zeros((10, 10, 10, 5))
+        pam.odf = np.zeros((10, 10, 10, sphere.vertices.shape[0]))
+
+        if not have_tables:
+            npt.assert_raises(TripWireError, save_peaks, fname, pam)
```

Comment:
Is the error handling in `_safe_save` tested elsewhere? Would you mind writing a test, if it isn't?